staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -356,6 +356,7 @@ func (h *UpgradeAwareHandler) tryUpgrade(w http.ResponseWriter, req *http.Reques
 		if err != nil && !strings.Contains(err.Error(), "use of closed network connection") {
 			klog.Errorf("Error proxying data from client to backend: %v", err)
 		}
+		writer.Close()
```

Comment:
backendConn is already closed above:
```
defer backendConn.Close()
```

we only need to do this if we construct a new flowrate writer, and it should probably be done in a defer

Reply:
Actually, calling `Close()` on the flowrate writer will also close the underlying writer, which is our backendConn, which will also close our reader. We don't want to do that.